### DISCUSSION TITLE
can we freeze the time for new transactions

### DISCUSSION THREAD

**Discussion by mookim-eth** [ConsenSys-archive/ganache#4027] [Q&A] [answered]
Some networks like Celo, require price oracle report to keep basic tokens functioning. In this case, as new transactions are minted in a block whose timestamp is set to current time, the forking cannot be used after serveral minutes. (error like this https://github.com/moolamarket/moola-fix/issues/3 )

Can we just freeze the time, aka set the same timestamp (can be the time of forking block) for new blocks?

**Comment by MicaiahReid** [comment, marked as answer]
I'm not sure I understand exactly what you need to do, but maybe this pattern will help:
```TypeScript
// ...ganache provider is already instantiated with startup options...

await provider.send("miner_stop"); // stop automatic mining so we can control when blocks are mined

// ...send whatever transactions you need in the block...

const timestamp = Date.now(); // set to whatever timestamp you want the block to have
await provider.send("evm_mine", [{timestamp: timestamp}] ); // mine a block with whatever timestamp you need

```

Does this help for your use-case?

**Reply by davidmurdoch** [reply]
You also might be able to make use of the `--miner.timestampIncrement` option and/or `--chain.time` to do what you want.
